common/quotas/priority_rate_limiter_impl_test.go

```diff
@@ -0,0 +1,428 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package quotas
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type (
+	priorityStageRateLimiterSuite struct {
+		suite.Suite
+		*require.Assertions
+
+		controller              *gomock.Controller
+		highPriorityRateLimiter *MockRateLimiter
+		lowPriorityRateLimiter  *MockRateLimiter
+		highPriorityReservation *MockReservation
+		lowPriorityReservation  *MockReservation
+		highPriorityAPIName     string
+		lowPriorityAPIName      string
+
+		rateLimiter *PriorityRateLimiterImpl
+	}
+)
+
+func TestPriorityStageRateLimiterSuite(t *testing.T) {
+	s := new(priorityStageRateLimiterSuite)
+	suite.Run(t, s)
+}
+
+func (s *priorityStageRateLimiterSuite) SetupSuite() {
+
+}
+
+func (s *priorityStageRateLimiterSuite) TearDownSuite() {
+
+}
+
+func (s *priorityStageRateLimiterSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+
+	s.controller = gomock.NewController(s.T())
+	s.highPriorityRateLimiter = NewMockRateLimiter(s.controller)
+	s.lowPriorityRateLimiter = NewMockRateLimiter(s.controller)
+	s.highPriorityReservation = NewMockReservation(s.controller)
+	s.lowPriorityReservation = NewMockReservation(s.controller)
+
+	s.highPriorityAPIName = "high-priority"
+	s.lowPriorityAPIName = "low-priority"
+	apiToPriority := map[string]int{
+		s.highPriorityAPIName: 0,
```

Comment:
can we test with non-contiguous priorities (such as 100, 200), or is that not supported?

Reply:
priority number being `non-contiguous` or `contiguous` is not important here